lib/foreman_puppet/engine.rb

```diff
@@ -1,5 +1,24 @@
 module ForemanPuppet
   class Engine < ::Rails::Engine
+    config.before_configuration do
+      unless ForemanPuppet.extracted_from_core?
+        require 'graphql'
+
+        module BaseObjectClassMethodPath
+          def field(*args, **kwargs, &block)
+            return if args.first == :environment && args.second.to_s == 'Types::Environment'
+            return if args.first == :environments && args.second.node_type.to_s == 'Types::Environment'
+            return if args.first == :puppetclass && args.second.to_s == 'Types::Puppetclass'
+            return if args.first == :puppetclasses && args.second.node_type.to_s == 'Types::Puppetclass'
+
+            super
+          end
+        end
+
+        GraphQL::Types::Relay::BaseObject.extend(BaseObjectClassMethodPath)
+      end
+    end
+
```

Comment:
a hacky way to skip the core types when the plugin is installed and to avoid duplicates.... any better ideas?

Reply:
I agree, the priority should be to make sure the plugin works with the `puppet_extraction` branch

Reply:
I agree, let's not bother by making it work with 2.5 as GraphQL is not stable API, so it's reasonable to say it works only with 3.0, if you need it, just wait with installing the plugin :)